metadata/overlays/db/metadata-db-deployment.yaml

```diff
@@ -5,10 +5,12 @@ metadata:
   labels:
     component: db
 spec:
+  replicas: 1
   selector:
     matchLabels:
       component: db
-  replicas: 1
+  strategy:
```

Comment:
em. Technically, we should use `statefulset` for mysql. because pod under deployment is controlled by ReplicaSet and replacement will be created immediately if you do an upgrade.  

This workaround has some down side, existing pods will be killed before new ones created. The problem is these are not all DB layers. `Recreate` strategy may cause downtime to its dependencies like KFP or Katib components. 
It requires this layer has the resilience on db failure. 

I have not tested this and not sure if anything abnoarmal there.

Reply:
@Jeffwan if you see this PR's description, currently, if you try and restart the MySQL Deployment, it will hang forever.

This is because the old Pod wont give up the lock on the DB files, so the new one can never be "ready" unless you manually kill the old Pod.

Reply:
I saw the problem.. My asking is if you see abnormal on katib or KFP side when you did upgrade with `Recreate ` strategy because DB service will have down time. connection maybe lost and if application doesn't handle this well, it could be an issue..

Reply:
@Jeffwan, we don't really have a choice, there will have to be some downtime as only one MySQL instance can lock the DB files at a time.

We just need to design Katib / KFP to be able to handle short downtimes of MySQL (like using exponential-backoff retries)

Reply:
sounds good